### PR TITLE
Updates blueprints to follow ember-suave

### PR DESCRIPTION
**Pull request by GreatWizard**
I update some blueprints to fit the latest version of ember-suave.

### CODE REVIEW

File: blueprints/app/files/tests/helpers/start-app.js [hunk 1]
```diff
@@ -2,13 +2,15 @@ import Ember from 'ember';
 import Application from '../../app';
 import config from '../../config/environment';
 
+const { merge, run }= Ember;
```
Comment (rwjblue) [inline blueprints/app/files/tests/helpers/start-app.js]: needs a space before `=` here.

File: blueprints/app/files/app/router.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
 import Ember from 'ember';
 import config from './config/environment';
 
-const Router = Ember.Router.extend({
+const { Router } = Ember;
```
Comment (rwjblue) [inline blueprints/app/files/app/router.js]: I'd prefer to have this be `const { Router: EmberRouter } = Ember;` and use `Router` for the apps router.

**Comment by stefanpenner** [comment]
This good but is churn, should we wait until we land the ember es6 module API to change this or do we feel this is OK.

File: tests/fixtures/addon/with-blueprint-override/blueprints/component/files/__root__/__path__/__name__.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import Ember from 'ember';
 <%= importTemplate %>
-export default Ember.Component.extend({<%= contents %>
+const { Component } = Ember;
```
Comment (kellyselden) [inline tests/fixtures/addon/with-blueprint-override/blueprints/component/files/__root__/__path__/__name__.js]: This probably needs some newline padding.

File: tests/fixtures/addon/with-blueprint-override/blueprints/component/files/__root__/__path__/__name__.js [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
 <%= importTemplate %>
-export default Ember.Component.extend({<%= contents %>
+
+const { Component } = Ember;
+
+export default Component.extend({<%= contents %>
 });
-//generated component successfully
\ No newline at end of file
+//generated component successfully
```
Comment (kellyselden) [inline tests/fixtures/addon/with-blueprint-override/blueprints/component/files/__root__/__path__/__name__.js]: was this newline change intentional?

**Comment by nathanhammond** [comment]
Echoing @stefanpenner here.

We'll land this PR, but merge it at the same time as other major blueprint changes.

(See also: #6082)

File: blueprints/app/files/tests/helpers/module-for-acceptance.js [hunk 5]
```diff
@@ -11,12 +11,12 @@ export default function(name, options = {}) {
       this.application = startApp();
 
       if (options.beforeEach) {
-        return options.beforeEach.apply(this, arguments);
+        return options.beforeEach.apply(...arguments);
```
Comment (Turbo87) [inline blueprints/app/files/tests/helpers/module-for-acceptance.js]: this looks like a bug to me. `apply` should be removed too, right?

Comment (rwjblue) [inline blueprints/app/files/tests/helpers/module-for-acceptance.js]: This is wrong actually.  It is a bug in JSCS's rule.  If we switch from `options.beforeEach.apply(this, arguments);` to `options.beforeEach(...arguments);` the `this` context inside of `options.beforeEach` changes from the same thing as the current context to the POJO itself (which is incorrect).

Comment (rwjblue) [inline blueprints/app/files/tests/helpers/module-for-acceptance.js]: See https://github.com/DockYard/ember-suave/issues/109 for some additional details.

Comment (Turbo87) [inline blueprints/app/files/tests/helpers/module-for-acceptance.js]: I feel like this conversation has happened before... but you're obviously right!

**Comment by Turbo87** [comment]
@locks any opinions here? should the blueprints mimic the guides? do the guides use destructuring too?

**Comment by stefanpenner** [comment]
well more specifically the future change will be something like

`import Application from 'ember/application'` (or something, the RHS is yet to be decided.

My vote is that we don't apply this, until the module API is decided. Or we will just incur change for change sake.

**Comment by locks** [comment]
I essentially make @stefanpenner's words mine. The churn would be both in CLI source/blueprints, as well as any documentation that refers to it, like the Guides.

**Comment by nathanhammond** [comment]
Closing this as what the eventual solution will be is undetermined and will likely require a full rewrite of this PR. We really appreciate the effort @GreatWizard and maybe you can help us out on that eventual solution as well! <3

**Pull request closed without merging**